### ISSUE TITLE
Errors raised in spec files outside of describe blocks cause spec files not to run

### ISSUE THREAD

**Issue by cbrwizard** [jasmine/jasmine#939]
Hey, is anybody else getting this or this is specific to a project my company is working on?

Let's say we have this file:

``` coffeescript
describe 'Sagamore.Model', ->
  beforeEach ->
    @model = new Sagamore.Model

  describe '#parse', ->
    beforeEach ->
      @resp = {
        'a': 1
        'hasOne?': true
      }

    it 'removes "?" from keys', ->
      response = @model.parse(@resp)

      expect(response.hasOwnProperty('hasOne'))
        .toEqual true
      expect(response.hasOwnProperty('hasOne?'))
        .toEqual false
```

Contents are not important. It runs perfectly, test passes, etc. We have 867 specs in our project and all of them pass.
The problem is that if you add a line

```
throw Error()
```

to the beginning of this file, this and 12 other specs do not get run at all. Jasmine says that 854 (note the changed number) specs are passing, 0 failures.

Is it a bug? Is it something on our side? It caused us a problem where we had about 100 specs not being run without us noticing for a week because of an error inside a spec file outside of a describe block.

**Comment by slackersoft** [comment]
Jasmine does not currently handle exceptions entirely outside the scope of the suite being declared and run. If the exception is thrown within a `describe`, either at runtime as part of a synchronous spec or not, jasmine will catch it and show the error.

This is something I've seen asked for before, but we haven't really found a good way to handle exceptions at that level. I would be happy to review a pull request that addresses this.

**Comment by himajasuman** [comment]
Hi, I tried figuring it out. I am new to node and jasmine. Any suggestions on where to catch the error(I mean in which file) would really help.

**Comment by slackersoft** [comment]
@himajasuman If the error is being thrown from within a `describe` jasmine will catch it and report it as an error in your suite and you shouldn't have to do anything. If the error is being thrown outside of a `describe` jasmine has no way to know what to do about the error (and thus just lets it break), you would have to catch it in your code.

**Comment by sgravrock** [comment]
This already works in Node, and it should be fixed in browsers as part of 3.0.

**Comment by slackersoft** [comment]
This has been fixed in the 3.0-features branch and will be released with Jasmine 3.0. Please check it out there and give us any feedback you have.

Note that 3.0-features is still under development, so you might want to pin to a particular commit.

Closing this. Thanks for using Jasmine!
